numandev1/react-native-compressor
# #212 Cancel background upload

This pull request follows issue https://github.com/numandev1/react-native-compressor/issues/212.
Within `backgroundUpload` there is an AbortController listener that will listen for 'abort' events and call the `cancelUpload` for all pending uploads.

This code is far from done, but hopefully this will jump start the development of this missing feature. 💪🏻

It would be greatly appreciated if @numandev1 takes over from here.

## Summary


Being able to cancel an upload is an essential part to prevent leaks in your app.

One issue with this PR in current state is that uploading multiple files at once using an asyncQueue will only cancel the last concurrent upload; the rest will continue uploading. 
Running the asyncQueue with concurrencyLimit of 1 will also lose connection to subsequent uploads. This is not the case with other uploader logic, therefore it is not an issue with the asyncqueue.

## Changelog

FEATURE - Add AbortController's signal to `backgroundUpload`

## Test Plan



There's no server available so I did not test the implementation in the example app. However I did test it in my own app and it is working as I expect it; it throws a `[Error: Socket closed]` on Android and PASTE ERROR on iOS, which is to be expected

Steps to reproduce:
1. Implement an asyncqueue to upload multiple files simultaneously
2. Select multiple files
3. Tap 'upload'
4. Wait for upload to start
5. Tap 'cancel'
6. Confirm upload cancellation
7. See the appropriate logs for cancelling the upload (no progress updates, 'upload cancelled', etc.).
